### ISSUE TITLE
Helper render issue causes failure

### ISSUE THREAD

**Issue by dotLou** [lrills/helm-unittest#72]
Given the following `_helper.tpl`:

```golang
{{- define "ingressClass" -}}
  {{- $ingressClass := .Values.ingress.class -}}
  {{- if .Values.global -}}
    {{- if .Values.global.ingressClass -}}
      {{- $ingressClass = .Values.global.ingressClass -}}
    {{- end -}}
  {{- end -}}
  {{- printf "%s" $ingressClass -}}
{{- end -}}
```

I actually ran into this issue with helm `2.11` and it was fixed somewhere in `2.12` but not explicitly. My _guess_ is this: https://github.com/helm/helm/pull/5300

In any case, here's what that looks like inside `ingress.yaml`:
```yaml
kubernetes.io/ingress.class: {{ template "ingressClass" . }}
```

And here's the failure on `helm unittest $CHARTNAME`:

```shell

### Chart [ mychart ] ./charts/mychart

 FAIL  test ingress     charts/mychart/tests/ingress_test.yaml
        - should set ingress class to nginx by default
                Error: parse error in "mychart/templates/_helper.tpl": template: collections/templates/_helper.tpl:49: unexpected "=" in operand


Charts:      1 failed, 0 passed, 1 total
Test Suites: 1 failed, 0 passed, 1 total
Tests:       1 failed, 1 errored, 0 passed, 1 total
Snapshot:    0 passed, 0 total
Time:        26.892096ms

Error: plugin "unittest" exited with error
```

The implication is there's a scope issue where the `=` inside the nested if doesn't have access to the variable defined in the second line of the helper above. It's imperative that the `=` work without it being changed to `:=` because that would actually assign a _new_ variable, that wouldn't be available outside the if, and thus couldn't be printed on the second to last line.

tl;dr; I'm wondering if upgrading `sprig` might fix this, but otherwise there's something else at play and helm fixed it (though perhaps unintentionally, haha).

**Comment by dotLou** [comment]
I was able to actually get this working locally after running a `dep ensure -update` so I suspect there's a dependency (perhaps helm itself!) that causes this to happen here. Would you be open to a PR that simply updates all dependencies?

**Comment by dotLou** [comment]
So I'm still getting this issue with `0.1.5`. However, if I build locally from master I don't get the issue. I'm going to try to dig to find out why that is.

**Comment by dotLou** [comment]
From master, with `0.1.5`:

![image](https://user-images.githubusercontent.com/2522036/56500040-37e4d280-64d7-11e9-96be-7d0510793ed8.png)

```shell
╰─± cat ~/.helm/plugins/helm-unittest/plugin.yaml                          
name: "unittest"
version: "0.1.5"
usage: "unittest for helm charts"
description: "Unit test for helm chart in YAML with ease to keep your chart functional and robust."
ignoreFlags: false
command: "$HELM_PLUGIN_DIR/untt"
hooks:
  install: "$HELM_PLUGIN_DIR/install-binary.sh"
  update: "$HELM_PLUGIN_DIR/install-binary.sh"
```

However, if I just install from master on my own machine, it works:

![image](https://user-images.githubusercontent.com/2522036/56500054-4f23c000-64d7-11e9-8d1d-3b3a45df8b9a.png)

**Comment by dotLou** [comment]
Okay so, if I build the binary inside of linux (or inside a docker image, using `make dockerimage`) I get the issue still. However, if I build it on mac (`make dist`) and then copy it to linux (or inside docker), then it works...


```shell

╭─~/dev/elastic-charts ⎇ ci/add-helm-unit-testing ✘ ⛵ docker-desktop
╰─± docker run --rm -it -v $PWD:/workspace unit
/unittest # ./untt /workspace/charts/collections/

### Chart [ collections ] /workspace/charts/collections/

 PASS  test ingress     ../workspace/charts/collections/tests/ingress_test.yaml

Charts:      1 passed, 1 total
Test Suites: 1 passed, 1 total
Tests:       4 passed, 4 total
Snapshot:    0 passed, 0 total
Time:        269.69383ms

```

Something is odd with building on linux. I'm going to keep digging.

**Comment by dotLou** [comment]
ok so, I see the same issue with the `make dockerimage` step.

If, however, I update that Dockerfile to the following:

```dockerfile
FROM golang:1.12.4-alpine3.9 as ALPINE-BUILDER
RUN apk --no-cache add --quiet make curl
RUN curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
WORKDIR /go/src/github.com/lrills/helm-unittest/
COPY . .
RUN install -d /opt && make install HELM_PLUGIN_DIR=/opt

FROM alpine:3.9 as ALPINE
COPY --from=ALPINE-BUILDER /opt /opt

```
Then that binary works:

```shell
─± docker run --rm -it -v $PWD:/workspace docker.io/irills/helm-unittest:0.1.5 /opt/untt /workspace/__fixtures__/with-subchart

### Chart [ parent-chart ] /workspace/__fixtures__/with-subchart

 PASS  test deployment  workspace/__fixtures__/with-subchart/tests/deployment_test.yaml
2019/04/22 13:56:51 Warning: Merging destination map for chart 'parent-chart'. The destination item 'annotations' is a table and ignoring the source 'annotations' as it has a non-table value of: <nil>
 PASS  test ingress     workspace/__fixtures__/with-subchart/tests/ingress_test.yaml
 PASS  test service     workspace/__fixtures__/with-subchart/tests/service_test.yaml
 PASS  test deployment  workspace/__fixtures__/with-subchart/charts/child-chart/tests/deployment_test.yaml
 PASS  test ingress     workspace/__fixtures__/with-subchart/charts/child-chart/tests/ingress_test.yaml
 PASS  test service     workspace/__fixtures__/with-subchart/charts/child-chart/tests/service_test.yaml

Charts:      1 passed, 1 total
Test Suites: 6 passed, 6 total
Tests:       14 passed, 14 total
Snapshot:    2 passed, 2 total
Time:        196.799297ms
```

**Comment by dotLou** [comment]
oh, the unit test CI steps use go 1.12 but the distribute uses go 1.10. That's the problem. PR incoming.

**Comment by guoyiang** [comment]
Is this issue going to be fixed? I ran into this issue recently as well.
I build the plugin with go after 1.12 and issue is not present. In CI config, test step uses go 1.12, but distribute step uses go 1.10.
Thanks! (And thanks for the info above)
